traitsui/editors/tuple_editor.py

```diff
@@ -216,21 +198,27 @@ def __init__(self, editor):
 
         self.view = View(Group(show_labels=(len_labels != 0), *content))
 
-    #-------------------------------------------------------------------------
-    #  Updates the underlying tuple when any field changes value:
-    #-------------------------------------------------------------------------
-
     def _field_changed(self, name, old, new):
         """ Updates the underlying tuple when any field changes value.
         """
+        editor = self.editor
+        value = editor.value
         index = int(name[1:])
         value = self.editor.value
         if new != value[index]:
-            self.editor.value = tuple(
-                [getattr(self, 'f%d' % i) for i in range(self.fields)])
+            new_value = tuple(
+                getattr(self, 'f{0}'.format(i)) for i in range(self.fields))
+            if self.fvalidate is not None:
+                if self.fvalidate(new_value):
```

Comment:
I am not convinced. If you want the validation to change the value then use a VaidatedTuple trait.  A number of editor already provide a format function to support conversions.